### PR TITLE
[mtg-135] add dao interaction check

### PR DESCRIPTION
**Pull request by kstepanovdev**

### CODE REVIEW

File: programs/voter-stake-registry/src/instructions/mod.rs [hunk 1]
```diff
@@ -31,6 +31,8 @@ mod unlock_tokens;
 mod update_voter_weight_record;
 mod withdraw;
 
+pub const DAO_PUBKEY: &str = "111111111111111111111111111111111";
```
Comment (StanChe) [inline programs/voter-stake-registry/src/instructions/mod.rs:34]: Is this something we'll need to define before deployment? Or is it a valid pub key already? We should add a comment or todo anyway here.

**Review by StanChe — APPROVED** [review]

File: programs/voter-stake-registry/src/instructions/mod.rs (hunk 1 above)
Comment (kstepanovdev) [reply programs/voter-stake-registry/src/instructions/mod.rs:34]: It's not valid for sure. It's a default value in fact. We need to provide the contract with the addr of our DAO to make sure `interaction-actions` are really coming from that DAO. I see no other way to do it except for hardcoding and I don't like it at all.

Comment (StanChe) [reply programs/voter-stake-registry/src/instructions/mod.rs:34]: Do we have the mainnet and devnet addresses already? We're not going to change the DAO addresses when we deploy staking, right?

Comment (kstepanovdev) [reply programs/voter-stake-registry/src/instructions/mod.rs:34]: We do, albeit they're not suitable for test cases. The address should be changed each time when the new test-dao is created.

**Pull request closed without merging**